### DISCUSSION TITLE
Different code with every 2nd and 3rd item in the collection (increment-tag, mod-modifier)

### DISCUSSION THREAD

**Discussion by BTB1999** [statamic/cms#7038] [Q&A] [answered]
Hi,

In my project, I have a collection of cases where the lay-out differs related to the position they're on. The first case should be full width, the second and third half width, then again full width, half width, half width, and so on.. It works when I hard code it, like this:

`{{ if count == 2 }} // do something {{ elseif count == 3 }} // do something else {{ else }} // do something else`

However, I want the now hard coded integers as variables. I've tried using `{{ increment from="2" by="3" }}`, which works, but using it in the if-statement like so: `{{ if count == {increment from="2" by="3"}`, always returns false.

Then I tried using the mod-modifier on the count like: `{{ count|mod:3}}` which in the loop outputs: "1, 2, 0, 1, 2, 0, etc...". This is exactly what I want, but when I do the following:

`{{ if {count|mod:3} == 1 }} // always false {{ elseif {count|mod:3} == 2 }} // always false {{ elseif {count|mod:3} == 0 }} // always true`

The `{{ if {count|mod:3} == 0 }}` always returns true. Is my syntax just not right, or is this a bug?

What is a good solution for this? I have quite a lot cases, so I don't want to put in all the right numbers of the 2nd and 3rd positions. Appreciate the help.

**Comment by JohnathonKoster** [comment, marked as answer]
Try surrounding the modifiers inside the conditions with parenthesis instead:

```antlers
{{ if (count|mod:3) == 1 }}

{{ elseif (count|mod:3) == 2 }}

{{ elseif (count|mod:3) == 0 }}

{{ /if }}
```

**Reply by BTB1999** [reply]
Thanks a lot, that's fixed it! A lot easier than expected
